### PR TITLE
refactor(modp2p): decouple Bitswap

### PR DESCRIPTION
**Pull request by Wondertan**
Decouples Bitswap into Client and Server. This is necessary to provide new custom options to the Client. In particular, this brings `client.WithoutDuplicatedBlockStats()` option to the table as nice little optimization for LNs. It disables one metrics we don't care much about atm, but avoids expensive Has call to DB as profiles show.

### CODE REVIEW

File: nodebuilder/p2p/bitswap.go [hunk 1]
```diff
@@ -29,16 +31,38 @@ const (
 // dataExchange provides a constructor for IPFS block's DataExchange over BitSwap.
 func dataExchange(params bitSwapParams) exchange.Interface {
 	prefix := protocol.ID(fmt.Sprintf("/celestia/%s", params.Net))
```
Comment (ramin) [inline nodebuilder/p2p/bitswap.go:33]: it'd be nice to have this as a func on bitSwapParams so you could spit out `params.ProtococID()` and we could write a little unit test for it, maybe even add `params.NeworkPrefix()` and we can test both and then just pass that as 3rd arg to `network.NewFromIpfsHost`

File: nodebuilder/p2p/bitswap.go [hunk 2]
```diff
@@ -29,16 +31,38 @@ const (
...
+
+	net := network.NewFromIpfsHost(params.Host, &routinghelpers.Null{}, network.Prefix(prefix))
+	srvr := server.New(
 		params.Ctx,
-		network.NewFromIpfsHost(params.Host, &routinghelpers.Null{}, network.Prefix(prefix)),
+		net,
 		params.Bs,
-		bitswap.ProvideEnabled(false),
+		server.ProvideEnabled(false), // we don't provide blocks over DHT
 		// NOTE: These below ar required for our protocol to work reliably.
```
Comment (ramin) [inline nodebuilder/p2p/bitswap.go:41]: nit: typo `ar` -> `are`

File: nodebuilder/p2p/bitswap.go [hunk 3]
```diff
@@ -29,16 +31,38 @@ const (
...
-		// See https://github.com/celestiaorg/celestia-node/issues/732
```
Comment (ramin) [inline nodebuilder/p2p/bitswap.go:38]: great history lesson for me 😸

**Review by ramin — DISMISSED** [review]

File: nodebuilder/p2p/bitswap.go (hunk 1 above)
Comment (Wondertan) [reply nodebuilder/p2p/bitswap.go:33]: I see your point. Not sure why that should be a method on bitSwapParams, though we could also extract that in a func and similarly unit test it. I don't think that the params struct needs any methods

Comment (Wondertan) [reply nodebuilder/p2p/bitswap.go]: Extract in a func

Comment (ramin) [reply nodebuilder/p2p/bitswap.go]: doing too many years of OOP is why i have the impulse to put it ON something, but func works exactly the same 👍

**Review by walldiss — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

**Pull request merged**